wire-ios-request-strategy/Sources/E2EIdentity/ACME/AcmeAPI.swift

```
@@ -0,0 +1,115 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+public protocol AcmeAPI {
+
+    func getACMEDirectory() async -> AcmeDirectoriesResponse?
+    func getACMENonce(url: String) async -> String?
+    func sendACMERequest(url: String, body: ZMTransportData) async -> ACMEResponse?
+    func sendChallengeRequest(url: String, body: Data?) async -> ChallengeResponse?
```

maybe we could throw errors instead of having optional return values